k8s/upgrade/src/bin/upgrade-job/helm/upgrade.rs

```diff
@@ -224,6 +238,68 @@ impl HelmUpgrade {
         HelmUpgradeBuilder::default()
     }
 
+    /// Use the HelmReleaseClient's upgrade dry runcommand to validate the upgrade command.
+    pub(crate) async fn dry_run(self) -> Result<()> {
+        let mut args_with_dry_run = self.core_chart_extra_args.clone();
+
+        if let Some(mut args) = args_with_dry_run {
+            args.push("--dry-run".to_string());
+            args_with_dry_run = Some(args);
+        } else {
+            args_with_dry_run = Some(vec!["--dry-run".to_string()]);
+        }
+
+        if self.chart_variant == HelmChart::Core && !self.already_upgraded {
```

Comment:
what does this chart variant mean?

Reply:
This is the helm chart variant of the helm chart installed in the cluster its either CORE or umbrella. @niladrih  would like to know your input for the difference between them, The dry run is only for the core chart in this PR.

Reply:
Ah I see...
And for umbrella we don't want to run this dry run?

Reply:
Cant confirm on exact reason but here https://github.com/openebs/mayastor-extensions/blob/f134161734b725202199c7911e64dd4dd4abb17c/k8s/upgrade/src/bin/upgrade-job/helm/upgrade.rs#L236 in case of Umbrella chart the upgrade is not being done . So i have skipped that in dry run as well. May be @niladrih  can help on this.

Reply:
hmm interesting then I wonder if we should suggest ppl to use the umbrella chart at all, seems better to just use mayastor chart directly so you can get upgrades like this.

Reply:
Hmm, if @niladrih  agrees to it. but that can be taken in another PR.